test/Kestrel.Interop.FunctionalTests/ChromeTests.cs

```diff
@@ -0,0 +1,96 @@
+    // https://github.com/aspnet/Extensions/issues/423
+    //[MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win81, SkipReason = "Missing Windows ALPN support: https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation#Support")]
+    [SkipIfChromeUnavailable]
+    public class ChromeTests : LoggedTest
+    {
+        private static readonly string _chromeArgs = "--headless --disable-gpu --allow-insecure-localhost --enable-logging --dump-dom --virtual-time-budget=10000";
+
+        [ConditionalTheory]
+        [InlineData("", "Interop HTTP/2 GET")]
+        [InlineData("?TestMethod=POST", "Interop HTTP/2 POST")]
+        public async Task Http2(string requestSuffix, string expectedResponse)
+        {
+            var deploymentParameters = new DeploymentParameters()
+            {
+                ApplicationPath = Path.Combine(TestResources.SolutionePath, "test", "Kestrel.Interop.TestSites"),
+                ServerType = ServerType.Kestrel,
+                TargetFramework = Tfm.NetCoreApp22,
+                RuntimeArchitecture = RuntimeArchitecture.x64
+            };
+
+            using (var deployer = new SelfHostDeployer(deploymentParameters, LoggerFactory))
+            {
+                var deploymentResult = await deployer.DeployAsync();
+                var requestUri = $"{deploymentResult.ApplicationBaseUri.Replace("[::]", "127.0.0.1")}";
+                var netLogPath = Path.GetTempFileName();
+                var startupLogPath = Path.GetTempFileName();
+                var shutdownLogPath = Path.GetTempFileName();
+                var logArgs = $"--log-net-log={netLogPath} --trace-startup --trace-startup-file={startupLogPath} --trace-shutdown --trace-shutdown-file={shutdownLogPath}";
+                var chromeArgs = $"{_chromeArgs} {logArgs} {requestUri}{requestSuffix}";
+                var chromeStartInfo = new ProcessStartInfo
+                {
+                    FileName = ChromeConstants.ExecutablePath,
+                    Arguments = chromeArgs,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true
+                };
+
+                Logger.LogInformation($"Staring chrome: {ChromeConstants.ExecutablePath} {chromeArgs}");
+
+                var headlessChromeProcess = Process.Start(chromeStartInfo);
+                var chromeOutput = await headlessChromeProcess.StandardOutput.ReadToEndAsync();
+
+                headlessChromeProcess.WaitForExit();
+
+                try
+                {
+                    Assert.Contains(expectedResponse, chromeOutput);
+                }
+                catch
+                {
+                    Logger.LogError("Test failed.");
+                    Logger.LogError("Chrome network logs:");
+                    Logger.LogError(File.ReadAllText(netLogPath));
```

Comment:
I'll update this to save these logs as separate files instead of writing everything into one file once https://github.com/aspnet/Logging/pull/906 is available.